src/main/java/application/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -21,20 +20,24 @@ public ShoppingCart create(ShoppingCart shoppingCart) {
 
     @Override
     public ShoppingCart addProduct(ShoppingCart shoppingCart, Product product) {
-        return shoppingCartDao.addProduct(shoppingCart, product);
+        Storage.getShoppingCarts().stream()
+                .filter(cart -> cart.equals(shoppingCart))
+                .forEach(cart -> cart.getProducts().add(product));
+        return shoppingCart;
```

Comment:
це можна спростити до:
```suggestion
        shoppingCart.getProducts().add(product);
        return shoppingCartDao.update(shoppingCart);
```
аналогічно з методом deleteProduct()

Reply:
done